[UI] polished navbar design with logo and grid styling

cwang12197 opened the pull request:
**[DESCRIPTION]**
- polished the header/navbar ui design with css grid styling, logo, title, and customized search bar 
<img width="1476" alt="Screen Shot 2022-07-25 at 1 16 54 AM" src="https://user-images.githubusercontent.com/68706299/180731360-43dc8b30-89d4-47ad-a657-a0395dd85f80.png">

cwang12197 commented:
**[DESCRIPTION]**

- able to display products that have been added to the basket in the "basket" section 
- alerts user if product selected is their own product (may change this to prevent it from showing in general)
- prevents users from adding the same product twice by only adding unique user ids

**[WORKING ON]**

- formatting the UI
- handling a delete button to remove it from the basket 
- adding an "on hold" which disables the product button for other users and adds an "on hold" alert

jennyhan1999 commented on food-waste-ui/src/components/Basket/Basket.jsx:
```diff
@@ -0,0 +1,39 @@
+    axios.get("http://localhost:3001/makeapost").then((response) => {
+      let allProducts = response.data.products;
+      let allProductsInBasket = [];
+      for (let i = 0; i < allProducts.length; i++) {
```
Is there a way we can do this without a nested for loop? Maybe storing another data struct? 

Is allProducts all the products the market place has within a specific radius? How would this work if we scale larger and many many products are available for users to choose from?

jennyhan1999 replied:
Is there a way we can do this without a nested for loop? Maybe storing another data struct? 

Is allProducts all the products the market place has within a specific radius? How would this work if we scale larger and many many products are available for users to choose from?

jennyhan1999 commented on food-waste-ui/src/components/Basket/Basket.jsx:
```diff
@@ -0,0 +1,39 @@
+      for (let i = 0; i < allProducts.length; i++) {
+        if (allProducts[i].basket) {
+          for (let j = 0; j < allProducts[i].basket.length; j++) {
+            if (allProducts[i].basket[j] == currentUserId) {
```
Naming is confusing, why is this checking if it == currentUserId?

jennyhan1999 replied:
Naming is confusing, why is this checking if it == currentUserId? And what is allProducts[i].basket representing?

jennyhan1999 commented on food-waste-ui/src/components/Map/Map.jsx:
```diff
@@ -78,9 +77,9 @@ export default function Map({
-  if (!isLoaded) {
-    return <Skeleton />;
-  }
+  // if (!isLoaded) {
```
nit: remove comments

jennyhan1999 replied:
nit: remove comments

jennyhan1999 commented on food-waste-ui/src/components/Map/Map.jsx:
```diff
@@ -6,15 +6,14 @@ import {
 import MapMarkers from "../MapMarkers/MapMarkers";
 import { Link } from "react-router-dom";
 import Geocoder from "react-native-geocoding";
+// import Loading from "../Loading/Loading";
```
nit: remove comments

jennyhan1999 replied:
nit: remove comments

jennyhan1999 commented on food-waste-express-api/routes/home.js:
```diff
@@ -13,4 +13,21 @@ router.get('/:objectId', async (req, res) => {
+        const Product = Parse.Object.extend("Products")
+        const query = new Parse.Query(Product)
+        query.get(productId).then((product) => {
+            product.addUnique("basket", currentUserId)
```
Could you go over how L20-27 works?

jennyhan1999 replied:
If this is casing on 'basket' maybe you can make this a global variable

jennyhan1999 commented on food-waste-ui/src/components/Loading/Loading.jsx:
```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
```
nice!

jennyhan1999 commented on food-waste-ui/src/components/Loading/Loading.jsx:
```diff
@@ -0,0 +1,30 @@
+
+("use strict");
+
+export default function Loading() {
```
nice!

jennyhan1999 commented on food-waste-ui/src/components/Navbar/Navbar.jsx:
```diff
@@ -39,51 +40,64 @@ export default function Navbar({
+        <SearchIcon className="search-icon" />
+      </div>
+      <div className="navbar-menu-items">
+        {screenWidth > sideBarWidth &&
```
Do you handle resizing the website window size? What about the mobile view?

jennyhan1999 commented on food-waste-express-api/routes/home.js:
```diff
@@ -13,4 +13,21 @@ router.get('/:objectId', async (req, res) => {
+router.post('/:objectId', async (req, res) => {
+    const productId = req.params.objectId;
+    let currentUserId = req.headers["current_user_id"]
+    try {
```
Could you go over how L20-27 works?

jennyhan1999 reviewed (commented):
It seems like you are doing some basket state handling as well but the commit title says these are just UI changes. You could separate this PR into two PRs for clarity

cwang12197 commented:
**[DESCRIPTION]** 

- added loading states when fetching data 
- added a remove icon in basket 

**[WORKING ON]**

- creating the onHold component

cwang12197 commented:
**[DESCRIPTION]**

- set products on hold and delete them from the market home page 

**[WORKING ON]**

- formatting the on hold 
- being able to remove items from on hold

jennyhan1999 reviewed (commented):
LGTM!

jennyhan1999 reviewed (commented):
nice!

jennyhan1999 commented on food-waste-express-api/routes/makeapost.js:
```diff
@@ -33,8 +33,14 @@ router.get('/', async (req, res) => {
+        let products = []
+
+        allProducts.map((product) => {
+            if (!product.get("placedOnHoldBy")) {
```
If you are using this input a lot of places maybe we can consider making it a global variable

jennyhan1999 commented on food-waste-ui/src/components/Basket/Basket.jsx:
```diff
@@ -0,0 +1,83 @@
+      let allProductsInBasket = [];
+
+      allProducts.map((product) => {
+        if (product.basket && product.basket.includes(currentUserId)) {
```
I feel this relationship is a little confusing naming wise

jennyhan1999 commented:
> 

Lets keep the PR comments simple:

##TITLE
Short title

##SUMMARY
Description of every change this PR has

##TEST PLAN
Unit test, screenshots, videos proving that this code has made the changes stated in the summary

cwang12197 commented:
##CREATION OF BASKET CLASS

##SUMMARY

-  redid basket component by storing products in a separate data structure in Parse
- added functions to fetch product and basket query for code modularity

cwang12197 commented:
REMOVAL OF PRODUCT FROM BASKET ON CLICK 

- removed product from database on click 
- cleaned up code (removed console.logs and //)

jennyhan1999 commented on food-waste-express-api/routes/home.js:
```diff
@@ -21,7 +21,6 @@ router.get("/addtobasket", async (req, res) => {
     try {
         let userBasket = await getUserBasket(currentUserId)
         let userBasketOfIds = await userBasket.get("basketOfProductId") //returns first instance
-        console.log("userBasketOfIds", userBasketOfIds)
```
Good to clean up these console.logs but please don't include them in your commits. Each commit should be free of console.logs + dead code + minimal comments. This makes it easier to review

jennyhan1999 commented on food-waste-ui/src/components/BasketCard/BasketCard.jsx:
```diff
@@ -1,14 +1,30 @@
+  const [location, setLocation] = React.useState("");
+
+  useEffect(() => {
+    console.log("product location", product.location);
```
cleanup

jennyhan1999 commented on food-waste-ui/src/components/Map/Map.jsx:
```diff
@@ -84,23 +85,17 @@ export default function Map({
-        setOrigin(addressComponent);
-      })
-      .catch((error) => console.warn(error));
+    if (!input) return;
```
Great job modularizing your code and creating appropriate helper functions!

cwang12197 commented:
TEST PLAN

This shows the removal of the product in the basket as well as the address (changed from geolocation to an actual location):


https://user-images.githubusercontent.com/68706299/181657517-5df42845-ca7b-486c-8b53-36b0b898e99b.mov

cwang12197 commented:
DISPLAY OF PRODUCTS ON HOLD

- able to show products user has put on hold in the "My Products on Hold" tab
- deletes products off market after putting on hold 
- clears basket after put on hold 

TEST PLAN

- compiles successfully with on hold products hidden on market 
<img width="1439" alt="Screen Shot 2022-07-29 at 2 07 37 AM" src="https://user-images.githubusercontent.com/68706299/181727205-5aa50b8b-d978-4bbd-83b4-07e5a2bd06e4.png">
<img width="405" alt="Screen Shot 2022-07-29 at 2 08 57 AM" src="https://user-images.githubusercontent.com/68706299/181727220-ab472473-1141-413a-a39a-186a9c8b0d09.png">
